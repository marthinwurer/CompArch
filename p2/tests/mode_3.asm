;
; tests addressing mode 3
;
; uses instructions MOV, HALT, BPT
;
; uses other addressing modes 0, 2
;
	.globl	start
ptrs:	.word	alpha,beta,gamma
. = . + 50
alpha:	.word	1
beta:	.word	2
gamma:	.word	4
;
. = . + 100
;
start:	mov	#ptrs,r2
	mov	#3,r1
	bpt
;
loop:	add	@(r2)+,r3
	bpt
	sub	#1,r1
	bgt	loop
;
quit:	bpt
	halt
;
	.end	start
