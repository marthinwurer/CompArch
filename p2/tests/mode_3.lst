      1                                 ;
      2                                 ; tests addressing mode 3
      3                                 ;
      4                                 ; uses instructions MOV, HALT, BPT
      5                                 ;
      6                                 ; uses other addressing modes 0, 2
      7                                 ;
      8 000000                                   .globl  start
      9 000000  000056' 000060' 000062' ptrs:    .word   alpha,beta,gamma
     10 000056                          . = . + 50
     11 000056  000001                  alpha:   .word   1
     12 000060  000002                  beta:    .word   2
     13 000062  000004                  gamma:   .word   4
     14                                 ;
     15 000164                          . = . + 100
     16                                 ;
     17 000164  012702  000000'         start:   mov     #ptrs,r2
     18 000170  012701  000003                   mov     #3,r1
     19 000174  000003                           bpt
     20                                 ;
     21 000176  063203                  loop:    add     @(r2)+,r3
     22 000200  000003                           bpt
     23 000202  162701  000001                   sub     #1,r1
     24 000206  003373                           bgt     loop
     25                                 ;
     26 000210  000003                  quit:    bpt
     27 000212  000000                           halt
     28                                 ;
     29 000214  000164'                          .end    start


alpha    000056 R  
beta     000060 R  
gamma    000062 R  
loop     000176 R  
pc     = 000007 %  
ptrs     000000 R  
quit     000210 R  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000164 RG 
