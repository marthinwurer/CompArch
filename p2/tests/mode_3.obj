000 6 2e 00 30 00 32 00
02e 2 01 00
030 2 02 00
032 2 04 00
074 4 c2 15 00 00
078 4 c1 15 03 00
07c 2 03 00
07e 2 83 66
080 2 03 00
082 4 c1 e5 01 00
086 2 fb 06
088 2 03 00
08a 2 00 00
074
