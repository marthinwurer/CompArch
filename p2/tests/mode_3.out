CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 164
000164:  PS[00]  012702  MOV (R7)+,R2
000170:  PS[04]  012701  MOV (R7)+,R1
000174:  PS[00]  000003  BPT
  R0[000000]  R1[000003]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000176]
000176:  PS[00]  063203  ADD ????,R3

Machine Halted - mode error

Simulated time 29 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
